pytorch_lightning/utilities/types.py

```diff
@@ -13,3 +14,9 @@
 _EVALUATE_OUTPUT = List[Dict[str, float]]  # 1 dict per DataLoader
 _PREDICT_OUTPUT = Union[List[Any], List[List[Any]]]
 _PARAMETERS = Iterator[torch.nn.Parameter]
+# TODO: expand allowed train_dataloaders types once fixed.
```

Comment:
which ones are not listed here? 
a todo for this PR or someone else?

Reply:
Depends on whether #7433 lands first or not